core/controllers/async_download.py

```diff
@@ -0,0 +1,142 @@
+import io
+from datetime import datetime
+from typing import IO
+
+from flask import current_app
+from notifications_python_client.notifications import NotificationsAPIClient
+from werkzeug.datastructures import FileStorage
+
+from config import Config
+from core.aws import _S3_CLIENT
+from core.controllers.download import download
+
+
+def async_download(
+    email_address: str,
+    file_format: str,
+    funds: list[str] | None = None,
+    organisations: list[str] | None = None,
+    regions: list[str] | None = None,
+    rp_start: str | None = None,
+    rp_end: str | None = None,
+    outcome_categories: list[str] | None = None,
+):
+    """Download data, store file in S3 and send an email to the user with the download link."""
+
+    response = download(
+        file_format=file_format,
+        funds=funds,
+        organisations=organisations,
+        regions=regions,
+        rp_start=rp_start,
+        rp_end=rp_end,
+        outcome_categories=outcome_categories,
+    )
+
+    if response.status_code != 200:
+        msg = f"Error downloading data from data-store/download endpoint. Status code: {response.status_code}"
+        current_app.logger.error(msg)
```

Comment:
For logging, we shouldn't format the error string up front, as it can mess with Sentry's ability to group errors. Feel free to see examples elsewhere in the app for how to structure log messages. Can I suggest something like this instead, using the `extra` dict from standard logging (note, `extra` dict can only contain basic data types, ie no full response object)?

```suggestion
        current_app.logger.error("Error downloading data from data-store/download endpoint. Status code: {response_status}", extra=dict(response_status=response.status_code))
```

Reply:
Changed as requested.